torch/csrc/distributed/autograd/init.cpp

```diff
@@ -28,23 +29,44 @@ PyObject* dist_autograd_init(PyObject* /* unused */) {
           .def(
               "_context_id",
               &DistAutogradContext::context_id,
-              py::call_guard<py::gil_scoped_release>());
+              py::call_guard<py::gil_scoped_release>())
+          .def("_send_functions", [](const DistAutogradContext& ctx) {
+            std::vector<py::object> funcs;
+            for (const auto& sendFunction : ctx.sendFunctions()) {
+              funcs.push_back(py::reinterpret_borrow<py::object>(
```

Comment:
I'm... pretty sure this is leaking Python objects. Inspection of `functionToPyObject` reveals that it incref objects before returning. `reinterpret_borrow` is the "copying" constructor: it doesn't take ownership of the input.

Reply:
(If I'm right, you should use `reinterpret_steal` here instead.)